scripts/app.js

```diff
@@ -0,0 +1,23 @@
+var app = angular.module('library', ["ui.router"]);
+app.config(function ($stateProvider, $urlRouterProvider) {
+    $stateProvider.state("loginPage", {
+        name: 'login',
+        url: '/login',
+        templateUrl: 'templates/login.html',
+        controller: 'validationController'
+    });
+      $stateProvider.state("bookShelf", {
+        name: 'bookShelf',
+        url: '/bookShelf',
+        templateUrl: 'templates/bookshelf.html',
+        controller: 'bookShelf'
+    });
+     $stateProvider.state("book", {
+        name: 'book',
+        url: '/bookShelf/book',
```

Comment:
url: /bookshelf/book/:bookId --> use stateparams in the resolve phase.
resolve : {
       book: [
            '$stateParams',
            '$q',
            'bookService'
            function($stateParams, $q, bookService) {
                    return book;// this book u need to inject in bookCtrl as a dependency
            }

}